libs/couch2pg/index.js

```diff
@@ -1,51 +1,36 @@
-// move data from couch to postgres.
-var pglib = require('pg-promise');
+var Promise = require('rsvp').Promise,
+    postgrator = require('postgrator'),
+    importer = require('./importer');
 
-// import the correct lib for web content:
-// travis couchdb does not support HTTPS, but prod systems require HTTPS
-var httplib = require('request');
+var COUCHDB_URL  = process.env.COUCHDB_URL,
+    POSTGRESQL_URL = process.env.POSTGRESQL_URL,
+    COUCH2PG_DOC_LIMIT = process.env.COUCH2PG_DOC_LIMIT;
 
-var Promise = require('../common').Promise;
+var couchdb = require('pouchdb')(COUCHDB_URL),
+    db = require('pg-promise')({ 'promiseLib': Promise })(POSTGRESQL_URL);
 
-var cdbfuncs = require('./cdbfuncs');
-var couchiter = require('./couchiter');
-var pgsql = require('./pgsql');
+var exports = module.exports = {};
+exports.import = function() {
+  return importer(db, couchdb, COUCH2PG_DOC_LIMIT).import();
+};
 
-module.exports = function () {
-  var sco;
+exports.migrate = function() {
+  return new Promise(function (resolve, reject) {
+    postgrator.setConfig({
+      migrationDirectory: __dirname + '/migrations',
+      schemaTable: 'couch2pg_migrations',
+      driver: 'pg',
+      connectionString: POSTGRESQL_URL
+    });
 
-  // establish a single connection to postgresql
-  return pglib({ 'promiseLib': Promise })(process.env.POSTGRESQL_URL)
-    .connect()
```

Comment:
i don't see `.connect()` or any task handling from pg-promise.

This means every `db.query` in `importer.js` is going to create a new connection. I count 2 DB connections for seq and the first delete, then 2*N DB connections as `loadAndStoreDocs` slams through N pages of docs (each page being no more than concurrentDocLimit).

Each iteration over `process.env.COUCH2PG_SLEEP_MINS` would open 2+2*N postgres DB connections.

This might be acceptable as long as N does not get large for too long. I was trying to keep each iteration sharing a single DB connection, but that is not done by default with pg-promise queries.